base/arrayshow.jl

```diff
@@ -457,6 +457,11 @@ typeinfo_eltype(typeinfo::Type{<:AbstractArray{T}}) where {T} = eltype(typeinfo)
 typeinfo_eltype(typeinfo::Type{<:AbstractDict{K,V}}) where {K,V} = eltype(typeinfo)
 typeinfo_eltype(typeinfo::Type{<:AbstractSet{T}}) where {T} = eltype(typeinfo)
 
+# types that can be parsed back accurately from their un-decorated representations
+function typeinfo_implicit(@nospecialize(T))
+    return T === Float64 || T === Int || T === Char || T === String || T === Symbol ||
```

Comment:
Is it for compiler friendlyness that you don't write something like `T in (Float64, Int, Char, String, Symbol)` ?